### ISSUE TITLE
make it easy to register new Archetype Catalog XML files on the fly via a ConfigMap

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-forge#509]
so that folks can dynamically add new archetypes to the fabric8-forge console.

Maybe fabric8-forge needs to watch ConfigMaps with a label and load them all in - then different ConfigMap resources could be for different archetype catalogs?

Then on the `Create App` -> `Create New App` -> `Quickstart` page, folks can then choose their custom archetype catalog in the combo box:

**Comment by jstrachan** [comment]
![fabric8-console](https://cloud.githubusercontent.com/assets/30140/18362565/20fdc4c0-75fe-11e6-9d79-661ecb2a14f1.png)

**Comment by rawlingsj** [comment]
Could we have a create 'new app creator' page that anyone could add an icon + base github repo + pipeline repo, that can be used to help people add any type / custom quickstarts?

**Comment by jstrachan** [comment]
yeah! We could make a slightly more generic flavor of the ipaas-quickstarts project that creates archetypes & a catalog for one or more projects. 'Adding a new project' would just add the git repo URL & name to some source file in the catalog builder project.

Though TBH I raised this issue initially as a way for folks to create ConfigMaps to publish additional archetype catalogs into forge - however they are created. Eg to add/remove existing archetype like Karaf or WF Swarm. 

Though making it easier to create archetypes new too sounds good! ;)

**Comment by rawlingsj** [comment]
Ah I see - yeah that would be good too ;)

I can definitely see enterprise customers wanting greater control to add and especially remove existing archetypes.

**Comment by gastaldi** [comment]
That could be created as a Forge command in the existing addons and use the same backend 👍

**Comment by jstrachan** [comment]
@gastaldi agreed! I was thinking of a forge addon that could query different plugins (ConfigMaps on kubernetes or file system or whatever) to find dynamic archetype catalogs to add on the fly etc

**Comment by gastaldi** [comment]
Forge has the ArchetypeCatalogFactoryRegistry service which makes it easy to register new ArchetypeCatalogs

**Comment by jstrachan** [comment]
@gastaldi ah thanks! Yeah I was just thinking we'd have a simple addon which watches for labelled/annotated `ConfigMaps` and uses the `ArchetypeCatalogFactoryRegistry` to add/remove catalogs dynamically on the fly etc. We can then add a forge command so folks can add/remove them to a cluster (which would under the covers write to the `ConfigMap`
